### DISCUSSION TITLE
Execute  Operation

### DISCUSSION THREAD

**Discussion by Art666OTS** [MarimerLLC/csla#2545] [Questions] [answered]
Hi All,

Nearly finished my upgrade and I got to the Execute operation. I am using the  EncapsulatedInvokeDto model.
I like to create a new file when looking at the code so that I can understand all that is happening. In this case I am using the OrderShipper class file as a starting point which I have renamed ArticleChanges class file. Apart from changes to names internally I think that they are identical. However I need to disable the CSLA0011 warning on the line 

var cmd = new ArticleChanges(orderId);

to get my code to compile and execute correctly. This is obviously not the right solution as it is not needed in the sample code. I have missed some subtlety in the code. Below is my code followed by the working code.

using Csla;
using SimpleDAL;
using System;

namespace SimpleBusinessLibrary
{
    [Serializable]
    public class ArticleChanges : CommandBase<ArticleChanges>
    {
        public static readonly PropertyInfo<int> IdProperty = RegisterProperty<int>(c => c.Id);
        public int Id
        {
            get { return ReadProperty(IdProperty); }
            private set { LoadProperty(IdProperty, value); }
        }

        public static readonly PropertyInfo<int> ShippingNumberProperty = RegisterProperty<int>(c => c.ShippingNumber);
        public int ShippingNumber
        {
            get { return ReadProperty(ShippingNumberProperty); }
            private set { LoadProperty(ShippingNumberProperty, value); }
        }

        public ArticleChanges()
        { /* required for MobileFormatter */ }

        private ArticleChanges(int orderId)
        {
            Id = orderId;
        }

        public static ArticleChanges ShipOrder(int orderId)
        {
#pragma warning disable CSLA0011 // Find CSLA Business Objects That Are Created Outside of a ObjectFactory
            var cmd = new ArticleChanges(orderId);
#pragma warning restore CSLA0011 // Find CSLA Business Objects That Are Created Outside of a ObjectFactory
            ArticleChanges ac = DataPortal.Execute<ArticleChanges>(cmd);
            // could also see shippingnumber here if needed.
            int sn = ac.ShippingNumber;
            return ac;
        }

        protected override void DataPortal_Execute()
        {
             var dal = DalFactory.GetDal<SimpleDAL.Interface.IArticle>();
             ShippingNumber = dal.ShipOrder(Id);
            // do something with shipping number if required
        }
    }
}

Working code

using Csla;
using System;

namespace Library
{
    [Serializable]
  public class OrderShipper : CommandBase<OrderShipper>
  {
    public static readonly PropertyInfo<int> OrderIdProperty = RegisterProperty<int>(c => c.OrderId);
    public int OrderId
    {
      get { return ReadProperty(OrderIdProperty); }
      private set { LoadProperty(OrderIdProperty, value); }
    }

    public static readonly PropertyInfo<int> ShippingNumberProperty = RegisterProperty<int>(c => c.ShippingNumber);
    public int ShippingNumber
    {
      get { return ReadProperty(ShippingNumberProperty); }
      private set { LoadProperty(ShippingNumberProperty, value); }
    }

    public OrderShipper()
    { /* required for MobileFormatter */ }

    private OrderShipper(int orderId)
    {
      OrderId = orderId;
    }

    public static void Ship(int orderId, EventHandler<DataPortalResult<OrderShipper>> callback)
    {
      var cmd = new OrderShipper(orderId);
      DataPortal.BeginExecute<OrderShipper>(cmd, callback);
    }

#if !SILVERLIGHT
    public static OrderShipper ShipOrder(int orderId)
    {
      var cmd = new OrderShipper(orderId);
      return DataPortal.Execute<OrderShipper>(cmd);
    }

    protected override void DataPortal_Execute()
    {
      using (var dalManager = DataAccess.DalFactory.GetManager())
      {
        var dal = dalManager.GetProvider<DataAccess.IOrderDal>();
        ShippingNumber = dal.ShipOrder(OrderId);
      }
    }
#endif
  }
}

Any assistance would be greatly appreciated.

**Comment by kcabral817** [comment, marked as answer]
Put this in ArticleChanges
```
[Create]
private void Create(int orderId)
{
    OrderId= orderId;
}
```
Then in the class you are calling or instantiating ArticleChanges (ShipOrder) put this:

```
var cmd = DataPortal.Create<ArticleChanges>(orderId);
cmd = DataPortal.Execute(cmd);
```

**Reply by Art666OTS** [reply]
Thanks for your quick response.
It has worked like a charm.

I can see how this is what should have been done to fit with the established pattern.

Thanks again.
Arthur
